Fix automake issues with having two Main.cc
Fix building with --as-needed

https://bugs.gentoo.org/358557

Patch provided by xarthisius@gentoo.org

--- src/Makefile.am
+++ src/Makefile.am
@@ -22,7 +26,7 @@
 $(top_srcdir)/src/config/Run.cc \
 $(top_srcdir)/src/config/Functions.cc \
 $(top_srcdir)/src/config/Functions.h \
-$(top_srcdir)/src/config/Main.cc \
+$(top_srcdir)/src/config/Main-config.cc \
 $(UTILS)
 
 UTILS = \
@@ -36,9 +40,11 @@
 $(top_srcdir)/src/utils/OptParser.h \
 $(top_srcdir)/src/utils/i18n.h
 
-wbar_CPPFLAGS = @modules_CFLAGS@ -I$(top_srcdir)/src/utils
-wbar_LDFLAGS = @modules_LIBS@
+AM_CPPFLAGS = -I$(top_srcdir)/src/utils
 
-wbar_config_CPPFLAGS = @gui_modules_CFLAGS@ -I$(top_srcdir)/src/utils
-wbar_config_LDFLAGS = @gui_modules_LIBS@
+wbar_CXXFLAGS = @modules_CFLAGS@
+wbar_LDADD = @modules_LIBS@
+
+wbar_config_CXXFLAGS = @gui_modules_CFLAGS@
+wbar_config_LDADD = @gui_modules_LIBS@
